hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -827,8 +827,9 @@ protected void putAllInternal(final Map<? extends Object, ? extends Object> entr
         int partitionCount = partitionService.getPartitionCount();
         boolean tooManyEntries = entries.size() > (partitionCount * factor);
         try {
+            List<Future<?>> futures = new LinkedList<Future<?>>();
+
```

Comment:
I created a new PR dealing with the strange behavior of this method:

https://github.com/hazelcast/hazelcast/pull/6094

Could you have a look at it?